scripts/pre-deploy-check.sh

```
@@ -0,0 +1,34 @@
+# 1. Verificar erros de sintaxe JSX/TSX
+echo "🔍 Verificando lint..."
+npm run lint
+if [ $? -ne 0 ]; then
+    echo "❌ Erro no lint detectado. Abortando."
+    exit 1
+fi
+
+# 2. Verificar tipos TypeScript
+echo "🔍 Verificando tipos TypeScript..."
+npm run type-check
+if [ $? -ne 0 ]; then
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.

(SC2181)